service/history/replicationTaskFetcher.go

```diff
@@ -192,8 +191,8 @@ Loop:
 				// We don't receive tasks from previous fetch so processors are all sleeping.
 				f.logger.Debug("Skip fetching as no processor is asking for tasks.")
 				timer.Reset(backoff.JitDuration(
-					time.Duration(f.config.AggregationIntervalSecs)*time.Second,
```

Comment:
can we move logic within the for loop into smaller functions?

Reply:
done :)